frontend/src/components/assessment-creation/FilterMenu.tsx

```diff
@@ -1,12 +1,130 @@
-import React from "react";
-import { Button } from "@chakra-ui/react";
+import React, { SetStateAction } from "react";
+import {
+  Button,
+  Text,
+  Popover,
+  PopoverBody,
+  PopoverContent,
+  PopoverFooter,
+  PopoverTrigger,
+  Divider,
+  Flex,
+} from "@chakra-ui/react";
+import { Select } from "chakra-react-select";
 import { FunnelIcon } from "../../assets/icons";
+import { MultiOption } from "../../types/SelectInputTypes";
 
-const FilterMenu = (): React.ReactElement => {
+export type FilterMenuProp = {
+  label: string;
+  setState: React.Dispatch<SetStateAction<Array<string>>>;
+  options: MultiOption[];
+};
+
+interface FilterMenuProps {
+  filterProps: FilterMenuProp[];
+}
+
+function addFilter(
```

Comment:
can we use `const` function notation here? explanation [linked](https://softwareengineering.stackexchange.com/questions/364086/why-use-const-foo-instead-of-function-foo)!

Reply:
Didn't know that, changed!